pkg/controller/sync/dispatch/managed.go

```diff
@@ -276,12 +281,21 @@ func (d *managedDispatcherImpl) recordVersion(clusterName, version string) {
 	d.versionMap[clusterName] = version
 }
 
-func (d *managedDispatcherImpl) StatusMap() status.PropagationStatusMap {
+func (d *managedDispatcherImpl) setResourcesUpdated() {
+	d.Lock()
+	defer d.Unlock()
+	d.resourcesUpdated = true
```

Comment:
This should be cleared otherwise I think we'll continually update status on every reconciliation, when we've already updated resources once and we have at least one status in the collection.

Reply:
A dispatcher's state is specific to the reconcile of a single federated resource, so it is safe not to clear this.

Reply:
Ah ok, I thought it might be relying on the default boolean zero value on every reconciliation. Maybe just a comment to clarify?

Reply:
It is an implicit invariant of `ManagedDispatcher` that it will not persist across reconciles (as per it only being [assigned to a local variable in a function called by reconcile](https://github.com/kubernetes-sigs/kubefed/blob/master/pkg/controller/sync/controller.go#L305)). The collection of the rest of cluster propagation status similarly depends on this invariant. Any future maintenance that missed this detail would result in broken tests.